source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp

```diff
@@ -0,0 +1,1225 @@
+                glUniformMatrix4fv(location, 1, transpose, mvm);
+            }
+        }
+    }
+
+    // Bind projection matrices
+    //
+    std::vector<std::string> projectionMatrixVariables =
+    {
+        "u_projectionMatrix",
+        //"u_projectionInverseMatrix",
+        "u_projectionTransposeMatrix",
+        //"u_projectionInverseTransposeMatrix",
+    };
+    for (auto projectionMatrixVariable : projectionMatrixVariables)
+    {
+        programInput = _uniformList.find(projectionMatrixVariable);
+        if (programInput != _uniformList.end())
+        {
+            location = programInput->second->_location;
+            if (location >= 0)
+            {
+                bool transpose = (projectionMatrixVariable.find("Transpose") != std::string::npos);
+                glUniformMatrix4fv(location, 1, transpose, pm);
+            }
+        }
+    }
+
+    // Bind view related matrices
+    std::vector<std::string> viewMatrixVariables =
+    {
+        //"u_viewMatrix",
+        //"u_viewInverseMatrix",
+        //"u_viewTransposeMatrix",
+        //"u_viewInverseTransposeMatrix",
+        "u_viewProjectionMatrix"
+        //"u_worldViewProjectionMatrix"
+    };
+    for (auto viewMatrixVariable : viewMatrixVariables)
+    {
+        programInput = _uniformList.find(viewMatrixVariable);
+        if (programInput != _uniformList.end())
+        {
+            location = programInput->second->_location;
+            if (location >= 0)
+            {
+                glUniformMatrix4fv(location, 1, GL_FALSE, pm);
+            }
+        }
+    }
+    
+    checkErrors();
+}
+
+void GlslValidator::findProgramInputs(const std::string& variable, 
+                                         const ProgramInputMap& variableList,
+                                         ProgramInputMap& foundList,
+                                         bool exactMatch)
+{
+    foundList.clear();
+
+    // Scan all attributes which match the attribute identifier completely or as a prefix
+    //
+    int ilocation = UNDEFINED_OPENGL_PROGRAM_LOCATION;
+    auto programInput = variableList.find(variable);
+    if (programInput != variableList.end())
+    {
+        ilocation = programInput->second->_location;
+        if (ilocation >= 0)
+        {
+            foundList[variable] = programInput->second;
+        }
+    }
+    else if (!exactMatch)
+    {
+        for (programInput = variableList.begin(); programInput != variableList.end(); programInput++)
+        {
+            const std::string& name = programInput->first;
+            if (name.compare(0, variable.size(), variable) == 0)
+            {
+                ilocation = programInput->second->_location;
+                if (ilocation >= 0)
+                {
+                    foundList[programInput->first] = programInput->second;
+                }
+            }
+        }
+    }
+}
+
+bool GlslValidator::bindAttribute(const GLfloat* bufferData,
```

Comment:
Binding of attribute streams. Builds shared h/w buffers for attribute types with multiple locations. May change this so we can tell the difference between one uv/color set and another. Lower priority for now.